src/pybind/mgr/cephadm/module.py

```diff
@@ -462,6 +464,12 @@ def _get_cephadm_service(self, service_type: str) -> CephadmService:
         assert service_type in ServiceSpec.KNOWN_SERVICE_TYPES
         return self.cephadm_services[service_type]
 
+    def _get_cephadm_binary_path(self) -> str:
+        import hashlib
+        m = hashlib.sha256()
+        m.update(self._cephadm.encode())
+        return f'/var/lib/ceph/cephadm/{m.hexdigest()}/cephadm'
```

Comment:
Do we want to still include the fsid between ceph and cephadm in the file path here?

Reply:
my reasoning was: independent of the FSID, the cephadm binary will be equal for each cluster

Reply:
That makes sense. Are we planning to start using /var/lib/ceph as a general location for ceph files like /etc/ceph? I can't remember anything being stored in here outside of an fsid directory before.

Reply:
> Are we planning to start using /var/lib/ceph as a general location for ceph files like /etc/ceph? 

I was thinking about `/etc/ceph`, but that is more meant for configuration files, not so much for binaries. 

> I can't remember anything being stored in here outside of an fsid directory before.

@pcuzner 's idea was to use `f'/var/lib/ceph/{fsid}/cephadm'`, but that is not safe when upgrading. Putting it outside of the `f'/var/lib/ceph/{fsid}'` has the potential to make things complicated, when garbage collecting cephadm binaries with multiple clusters running at the same time, but we have the same problem with garbage collecting images already.

Reply:
``/var/lib/ceph/{fsid}/cephadm/{hash}``?